ansible_base/resource_registry/rest_client.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from collections import namedtuple
+from urllib.parse import urlencode
+
+import requests
+import urllib3
+
+ResourceRequestBody = namedtuple("ResourceRequestBody", ["ansible_id", "service_id", "resource_type", "resource_data"], defaults=(None, None, None, None))
+
+
+urllib3.disable_warnings()
+
+logger = logging.getLogger('ansible_base.resources_api.rest_client')
+
+
+class ResourceAPIClient:
+    """
+    Client for Ansible services to interact with the service-index/ api
+    """
+
+    def __init__(self, service_url: str, service_path: str, requests_auth_kwargs: dict, verify_https=True):
+        """
+        service_url: fully qualified hostname for the service that the client
+            is connecting to (http://www.example.com:123).
+        service_path: path on the service where the service-index/ api is found
+            (/api/v1/service-index/).
+        requests_auth_kwargs: dictionary of additional args to pass to requests.request()
+            ({auth=("username", "password")})
+        """
+
+        self.base_url = f"{service_url}/{service_path.strip('/')}/"
+        self.requests_auth_kwargs = requests_auth_kwargs
+        self.verify_https = verify_https
+
+    def _make_request(self, method: str, path: str, data: dict = None) -> requests.Response:
+        url = self.base_url + path.lstrip("/")
+        logger.info(f"Making {method} request to {url}.")
+
+        kwargs = {**self.requests_auth_kwargs, "method": method, "url": url, "verify": self.verify_https}
+
+        if data:
+            kwargs["json"] = data
+
+        return requests.request(**kwargs)
+
+    def _get_filter_query_params(self, filters: dict):
```

Comment:
I probably should have thought of this in the first review, but maybe it makes more sense to just pass the filters in as `params` to `requests.request()` and not construct the querystring ourselves:

```python
In [5]: x = requests.request(url='https://example.com', params={'foo': 'bar', 'color': 'green'}, method='get')

In [6]: x.url
Out[6]: 'https://example.com/?foo=bar&color=green'
```

I think this handles encoding for us, too.

Reply:
huh. I didn't know you could do that. Thanks!